### PR TITLE
Break up 'How to use' into multiple pages and related updates

### PR DESCRIPTION
**Pull request by leoebfolsom**
The OS Diff "How to use" page was getting pretty long as I added examples and explanations, so I broke it up into three pages:

- Command line
- TOML Config file
- Python

A few other items that followed naturally from this initial/primary goal:

- I also added Snowflake-specific TOML config instructions to the "How to use > TOML Config file" page
- I moved the "Databases we support" to its own page--hopefully this will help some new users quickly answer questions about whether this will be useful to them out of the box.
- I created a new page, "How to use > Options" because this pertains to the other three "How to use" pages.
- Added config key syntax to Options page.

### CODE REVIEW

**Comment by leoebfolsom** [comment]
I think everything is failing because of something I didn't do right with the docusaurus structure and the how_to_use folder as a menu item. I'll dig into it tomorrow if no one has a quick solution they can point me to.

**Comment by leoebfolsom** [comment]
Thank you @erezsh, I like that direction, and I'll take a stab at it 🙌

**Comment by williebsweet** [comment]
> @williebsweet ow @dlawin would you please advise how to make the how_to_use menu item, which is a folder, render as "How to use"? Now that it's a folder, I haven't figured out how to specify the name. (I can dig more, but figured you might know.)
@leoebfolsom - [this commit](https://github.com/datafold/datafold-docs/pull/53/commits/821bbd2cf7412c38cb7d5a1862e30bf21abcd45a) addresses the question above

**Comment by leoebfolsom** [comment]
@erezsh I mocked up an options table. However, I wasn't sure what to put in the "config key" column. Would you please either:

- fill in that column
- point me to documentation
- let me know how to translate the CLI Switch to the syntax for config key?

I can guess, but don't want to waste my time making an assumption.

Thanks!

**Comment by erezsh** [comment]
Yes, that's exactly why we need this table :)

It basically all follows the same pattern - you take the longest form, and swap the `-` for `_`. So 

```
--key-columns   ->  key_columns
--materialize-all-rows     -> materialize_all_rows
```
and so on

**Comment by leoebfolsom** [comment]
@erezsh thanks! to be clear, the config keys are to be used in the TOML file?

Is there a third version for a Python use case, or does it leverage either the CLI switches or the config keys?

**Comment by dlawin** [comment]
minor but the wrapping of the long form flags can look odd:
Screen Shot 2022-11-09 at 10 36 49 AM

I _think_ it's possible to force them not to wrap

**Comment by erezsh** [comment]
Yes, Python and the TOML file share the same keys.

--conf and --run don't appear in the TOML or Python. But if you omit `--run` and define `run.default`, it's like calling `--run default`

**Comment by erezsh** [comment]
@dlawin I agree that they shouldn't wrap

File: docs/os_diff/how_to_use/options.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 4
+title: Options
+---
+
+
+You can use the following options to specificy the configuration of a data-diff run.
```
Comment (erezsh) [inline docs/os_diff/how_to_use/options.md:7]: Specify

File: docs/os_diff/how_to_use/options.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+- **Config Key**: Use this syntax when using a [TOML configuration file](./how_to_use_with_toml.md).
+```
```
Comment (erezsh) [inline docs/os_diff/how_to_use/options.md:10]: Add TOML highlighting

File: docs/os_diff/how_to_use/options.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  stats = true
+```
+
+- **CLI Switch**: Use this syntax when writing a [command line input](./how_to_use_with_command_line.md).
+```
+data-diff DB1_URI TABLE1_NAME DB2_URI TABLE2_NAME --debug --v -k order_id
+```
+
+
+| Config key     | CLI switch                 | Description                                      | Within-Database | Cross-Database |
```
Comment (erezsh) [inline docs/os_diff/how_to_use/options.md:23]: Fix wrapping of switches causing artifacts like

```
   --i or --
   interactive
```

File: docs/os_diff/how_to_use/options.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+| `debug`   | `-d` or `--debug`           | Print debug info. | ✅ | ✅ |
+| `verbose`   | `-v` or `--verbose`           | Print extra info. | ✅ | ✅ |
+| `interactive`   | `-i` or `--interactive`           | Confirm queries, implies `--debug` | ✅ | ✅ |
+| `json`   | `--json`           | Print JSONL output for machine readability. | ✅ | ✅ |
+| `min_age`   | `--min-age`       | Considers only rows older than specified. Useful for specifying replication lag. Example: `--min-age=5min` ignores rows from the last 5 minutes. Valid units: `d, days, h, hours, min, minutes, mon, months, s, seconds, w, weeks, y, years` | ✅ | ✅ |
+| `max_age`   | `--max-age`           | Considers only rows younger than specified. See `--min-age`. | ✅ | ✅ |
+| `threads`   | `-j` or `--threads`           | Number of worker threads to use per database. Default=1. | ✅ | ✅ |
+| `where`   | `-w`, `--where`           | An additional 'where' expression to restrict the search space. | ✅ | ✅ |
+| `conf`, `run`   | `--conf`, `--run`           | Specify the run and configuration [from a TOML file](how_to_use_with_toml.md). | ✅ | ✅ |
```
Comment (erezsh) [inline docs/os_diff/how_to_use/options.md:39]: no equivalent config key

File: docs/os_diff/how_to_use/options.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+| `bisection_factor`   | `--bisection-factor`           | Segments per iteration. When set to 2, it performs binary search. |   | ✅ |
+| `materialize`   | `-m`, `--materialize`           | Materialize the diff results into a new table in the database. If a table exists by that name, it will be replaced. Use `%t` in the name to place a timestamp. Example: `-m test_mat_%t` | ✅ |  |
+| `assume_unique_key`   | `--assume-unique-key`           | Skip validating the uniqueness of the key column during joindiff, which is costly in non-cloud dbs. | ✅ |  |
+| `sample_exclusive_rows`   | `--sample-exclusive-rows`           | Sample several rows that only appear in one of the tables, but not the other. Use with `-s`. | ✅ |  |
+| `materialize_all_rows`   | `--materialize-all-rows`           | Materialize every row, even if they are the same, instead of just the differing rows. | ✅ |  |
+| `table_write_limit`   | `--table-write-limit`           | Maximum number of rows to write when creating materialized or sample tables, per thread. Default=1000. | ✅ |  |
+| `algorithm`   | `-a`, `--algorithm`  | Force algorithm choice. Options: `auto`, `joindiff`, `hashdiff` | ✅ |  |
```
Comment (erezsh) [inline docs/os_diff/how_to_use/options.md:48]: algorithm is a general swithch, not just for joindiff

**Review by erezsh — CHANGES_REQUESTED** [review]
I like the direction.

I still think it would help to split them to categories. Like 'diff-related', 'table filtering', 'display options', 'misc'

File: docs/os_diff/how_to_use/options.md (hunk 3 above)
Comment (leoebfolsom) [reply docs/os_diff/how_to_use/options.md:23]: @dlawin @erezsh I found a lot of discussion online about how the way GitHub works with markdown, there's not a great option for wrapping. I found a hacky solution that I hope you both can tolerate unless you can point me to a better option. commit forthcoming but it'll look like this:

```
| Config key | CLI&nbsp;switch&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| Description | Within-Database | Cross-Database |
```

File: docs/os_diff/how_to_use/options.md (hunk 2 above)
Comment (leoebfolsom) [reply docs/os_diff/how_to_use/options.md:10]: Sorry, what do you mean by this?

**Comment by leoebfolsom** [comment]
> I still think it would help to split them to categories. Like 'diff-related', 'table filtering', 'display options', 'misc'

@erezsh I tried to get at this with the additional columns but I see what you mean--let me mock something up.

**Comment by leoebfolsom** [comment]
@erezsh I took some liberties with the categories. For one, I didn't think "diff-related" was meaningful because it's all basically diff-related. See what you think. I'm open to your suggestions.

I added an additional column, which is more straightforward than getting github+markdown to render subheads. Another option would be to have multiple tables, each dedicated to a category. I thought the approach I took was more straightforward for the reader (the categories aren't _that_ distinctive so as to require separate tables), but there are tradeoffs either way.

File: docs/os_diff/how_to_use/options.md (hunk 2 above)
Comment (erezsh) [reply docs/os_diff/how_to_use/options.md:10]:
```toml
[this.is_highlighted]
so_is = "this"
```

Comment (leoebfolsom) [reply docs/os_diff/how_to_use/options.md:10]: This feels like a riddle. Can you please explain more or give an example of what you would like me to add? Don't I already use both version in the example code snippet I included in the subsequent lines?

```
# Specify the default run parameters
[run.default]
  verbose = true
  stats = true
```

Sorry if I'm missing something obvious.

**Comment by erezsh** [comment]
Looking good! I like the Performance category.

But I would say sample_exclusive_rows is more output or settings than performance. Also I would put the first four in Schema instead of Settings.

**Comment by leoebfolsom** [comment]
> But I would say sample_exclusive_rows is more output or settings than performance. Also I would put the first four in Schema instead of Settings.

Totally agree--I misread the description earlier.

**Comment by erezsh** [comment]
You're still not highlighting the TOML - You're still not highlighting the TOML - https://deploy-preview-53--datafold-docs.netlify.app/os_diff/how_to_use/how_to_use_with_toml

Does  docosaurus not support it?

**Comment by leoebfolsom** [comment]
@erezsh as I said [here](https://github.com/datafold/datafold-docs/pull/53#discussion_r1018711622), I didn't understand what you meant by "highlighting the toml." I'd never heard of "highlighting" before, and in the code snippet you provided as an example, the fact that you included `toml` after the first three ticks wasn't apparent to me, so I still failed to understand. I only just now figured out what I think you want. I did this by clicking edit on your comment to see if there was something for me to discover. I'm going to try to do it now.

**Comment by leoebfolsom** [comment]
@erezsh I agree it looks much better with the highlighted toml. Thanks for making me aware of this option!

**Comment by erezsh** [comment]
I still don't see it highlighted in the preview

**Comment by leoebfolsom** [comment]
@erezsh bummer. Looks like it's supported and would have to be added to https://github.com/highlightjs/highlight.js/tree/main/src/languages .

**Comment by leoebfolsom** [comment]
@erezsh since there's not a short-term way of getting TOML highlighting, can you please approve or request additional changes?

**Review by erezsh — APPROVED** [review]

**Review by dlawin — APPROVED** [review]

**Comment by erezsh** [comment]
Makes me wonder why we aren't using a Python highlighter, because they won't have this limitation.

**Comment by leoebfolsom** [comment]
@erezsh do you mean using 
```
```python
```

instead of 

```
```toml
```

at the beginning of the TOML blocks? 

That seems to work (see this screenshot, from running the docs on localhost with that change):


Screen Shot 2022-11-11 at 9 33 58 AM

**Comment by erezsh** [comment]
Actually I meant using a Python code highlighter.

But your solution is a big improvement too!

**Pull request merged**